### DISCUSSION TITLE
how to use  ????

### DISCUSSION THREAD

**Discussion by caicaicai1996** [nirenjan/x52pro-linux#23] [Q&A]
I have installed the package,but i don't know how to use it to set .please help me ,thanks .

**Comment by nirenjan** [comment]
There is a command line utility `x52cli`, which should have been installed if you ran `sudo make install`. This utility controls the LED brightness, state (on X52 pro only), MFD text, brightness and clock. If you just built the source code, this will be under `utils/cli`. You may have to run this as root if you haven't installed the udev rules.

Refer to the comments in [this file](https://github.com/nirenjan/x52pro-linux/blob/65c889827a7fe1acc9bc8e58541a26a6bae35c2f/utils/cli/x52_cli.c) for the syntax

**Comment by caicaicai1996** [comment]
First of all, thank you very much for your reply!!! I've installed the first and third parts according to the installation instructions. The above picture is the situation after my installation. My lab wants to use X52 to control the ship.


The mapping of X52 key to keyboard has been completed in Windows system. Does your program support the same thing? Thanks again!!!





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "nirenjan/x52pro-linux"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年12月22日(星期二) 下午2:32
收件人:&nbsp;"nirenjan/x52pro-linux"<x52pro-linux@noreply.github.com&gt;;
抄送:&nbsp;"Song ZiCheng"<1412565924@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [nirenjan/x52pro-linux] how to use  ???? (#23)





 
There is a command line utility x52cli, which should have been installed if you ran sudo make install. This utility controls the LED brightness, state (on X52 pro only), MFD text, brightness and clock. If you just built the source code, this will be under utils/cli. You may have to run this as root if you haven't installed the udev rules.
 
Refer to the man page for the syntax (man x52cli.1)

**Reply by nirenjan** [reply]
As of now, this program doesn't yet support mapping buttons/axis to keyboard or mouse events. However, there is another open source tool called antimicro which might be able to implement this.

**Comment by caicaicai1996** [comment]
OK! Thank you very much. I will try to solve it. Greetings from China. I am a graduate student in mechanical engineering. But the students around, from the perspective of work, are self-taught programming, but the level is very limited. And the lack of innovation ability, is to copy other people's code. I'm learning C + +, there is no computer foundation, can you give some learning suggestions. Through the Internet, a boy from rural China came into contact with the world. Thanks again!!!!


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "nirenjan/x52pro-linux"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年12月22日(星期二) 下午3:56
收件人:&nbsp;"nirenjan/x52pro-linux"<x52pro-linux@noreply.github.com&gt;;
抄送:&nbsp;"Song ZiCheng"<1412565924@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [nirenjan/x52pro-linux] how to use  ???? (#23)





 
As of now, this program doesn't yet support mapping buttons/axis to keyboard or mouse events. However, there is another open source tool called antimicro which might be able to implement this.
